falcon/response.py

```diff
@@ -123,6 +133,103 @@ def set_stream(self, stream, stream_len):
+                negative integer.  After delta-seconds seconds elapse,
+                the client should discard the cookie.
+                A value of 0 means the cookie should be discarded immediately.
+            expires (str): Defines when the cookie should expire.
+                the default is to expire when the browser is closed.
+                The format given should be "Wdy, DD Mon YYYY HH:MM:SS GMT"
+            secure (bool): The Secure attribute directs the user
+                agent to use only secure means to contact the origin
+                server whenever it sends back this cookie.
+            version (str): The Version attribute, a decimal integer,
+                identifies to which version of the state management
+                specification the cookie conforms.
+            httponly (bool): The attribute httponly specifies that the cookie
+                is  only transferred in HTTP requests, and is not accessible
+                through JavaScript. This is intended to mitigate some forms
+                of cross-site scripting.
+
+        Note:
+            Kwargs and there valid values are all
+            specified in http://tools.ietf.org/html/rfc2109.html
+
+        """
+
+        if self._cookies is None:
+            self._cookies = SimpleCookie()
+
+        self._cookies[key] = value
+
+        if len(cookie_params) > 0:
+
+            # TODO(tbug): Okay, this looks a little sloppy, we
+            # since we can't easily use max-age in our kwargs we
+            # remap max_age to max-age here.
+            if "max_age" in cookie_params:
+                cookie_params["max-age"] = cookie_params["max_age"]
+                del cookie_params["max_age"]
+
+            for param, value in cookie_params.items():
+                self._cookies[key][param] = value
+
+    def get_cookie(self, name):
+        """Return cookie by name.
+
+        The cookie will be returned as a Morsel object.
+
+        Note:
+            In python 2;
+            https://docs.python.org/2/library/cookie.html#morsel-objects
+            In python 3;
+            https://docs.python.org/3/library/http.cookies.html#morsel-objects
+        """
+        if self._cookies is None:
+            return None
+        if name in self._cookies:
+            return self._cookies[name]
+        else:
+            return None
+
+    def get_cookies(self):
+        """Return all cookies as a list of Morsel objects
+
+        Note:
+            You cannot set response cookies by changing the returned list.
+            Use response.set_cookie() for that instead.
+        """
+
+        # NOTE(tbug): Not sure i like this function, but we _do_ need
+        # a way to get all cookies, right?
+        return list(self._cookies.values())
+
+    def unset_cookie(self, name):
+        """Unset a cookie from the response
+
+        """
+        if self._cookies is not None:
```

Comment:
There's a subtle inconsistency here. If no cookies were ever set, and I try to delete one, it's a noop. However, if one ore more cookies were set, but I try to unset a different cookie that doesn't exist, L231 will raise `KeyError`.